src/Task/domain/entity/Task.spec.ts

```diff
@@ -1,14 +1,25 @@
 import { Task } from './Task';
+import { SchedulingOption } from './TaskInterface';
 
 describe('Task Entity', () => {
-  it('should create a valid task', () => {
+  it('should create a valid task with flexible scheduling and timeslot', () => {
+    const timeslot = {
+      startTime: '10:00 AM',
+      endTime: '12:00 PM',
+    };
     const task = new Task({
       title: 'Sample Title',
       description: 'Sample Description',
       user: {
         userId: 'user123',
         fullName: 'John Doe',
       },
+      category: 'Sample Category',
+      location: 'Sample Location',
+      budget: 100,
+      scheduling: SchedulingOption.Flexible,
+      timeslot,
+      createdon: new Date(),
     });
 
     if (task.taskIsValid && typeof task.taskIsValid === 'function') {
```

Comment:
it is not the practice to put if else at unit test. test should also follow SRP

Reply:
fixed it, please take a look.